specification/quota/resource-manager/Microsoft.Quota/preview/2021-03-15-preview/quota.json

```diff
@@ -0,0 +1,1226 @@
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceNameInParameters"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "createQuotaRequest",
+            "in": "body",
+            "description": "Quota requests payload.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CurrentQuotaLimitBase"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Returns Quota requests details.",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestOneResourceSubmitResponse"
+            }
+          },
+          "201": {
+            "description": "The request is accepted and is being processed. Use requestId to check the status.",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestSubmitResponse201"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Quota/quotaLimits": {
+      "get": {
+        "tags": [
+          "QuotaInformation"
+        ],
+        "description": "Get a list of current service limits (quota) and usages of all the resources. The response from List API can be leveraged to submit quota update requests.",
+        "operationId": "Quota_List",
+        "x-ms-examples": {
+          "Quotas_listUsagesForCompute": {
+            "$ref": "./examples/getComputeUsages.json"
+          },
+          "Quotas_listUsagesMachineLearningServices": {
+            "$ref": "./examples/getMachineLearningServicesUsages.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Quota information will be returned as response with pagination.",
+            "schema": {
+              "$ref": "#/definitions/QuotaLimits"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Quota/quotaLimitsRequests/{id}": {
```

Comment:
>id [](http://example.com/codeflow?start=53&length=2)

How customers get this id?

Reply:
Also, how is a limit request created? I don't see any PUT api.

Reply:
The customer can get the Id from the Put /patch response - id field. It's also included in the URI for  Azure-AsyncOperation, Operation-Location headers, as Azure guidelines.
https://docs.microsoft.com/en-us/rest/api/reserved-vm-instances/quota/createorupdate#examples - The Id field in the response.

Reply:
The user can check the quota request status using /servicelimitsRequests/{requestId} Api. Similar to the operation -checkStatus suggested. A user can submit multiple quota requests for the same resource, so we are providing unique requestId to check the status of a specific request.
The put and Get APIs for quota is based on /servicelimits/{resourceName}.
The Get API gets the current quota for the resources.
The Put API submits a quota request for the resource.
The Put response returns the operationsStatus uri, as per Async operation guidelines. It also provides the Operation-Location header, retry-after. The response id field contains the resource uri for quota request.

The Azure SDK and deployment template uses the AsyncOperation, Operation-Location headers and retry-after. The SDK and deployment template returns the final status, when the operation has reached terminal state - success or failed.
We have documented the process for the User operations. The Azure portal also uses the similar mechanism.